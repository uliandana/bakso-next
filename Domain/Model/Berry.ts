export interface Berry {
  id: string,
  name: string,
  sprite: string,
  firmness: string,
}
