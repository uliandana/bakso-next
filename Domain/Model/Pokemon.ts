export interface Pokemon {
  id: string,
  name: string,
  nameSlug: string,
  sprite: string,
}
