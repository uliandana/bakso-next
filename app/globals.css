@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --foreground-rgb: 255, 255, 255;
  --background-start-rgb: 188,55,54;
  --background-end-rgb: 188,55,54;
  font-size: calc(768px * 16 / 375);
}

@media (prefers-color-scheme: dark) {
  :root {
    --foreground-rgb: 255, 255, 255;
    --background-start-rgb: 0, 0, 0;
    --background-end-rgb: 0, 0, 0;
  }
}

@layer utilities {
  .card-flip {
    position: absolute;
    backface-visibility: hidden;
    transform: rotateY(0);
    transition: transform 0.5s ease-in-out;
  }
  .card-flipped {
    position: absolute;
    backface-visibility: hidden;
    transform: rotateY(180deg);
    transition: transform 0.5s ease-in-out;
  }
  .silhouette {
    filter: brightness(0) drop-shadow(0.0625rem 0.125rem 0.25rem rgb(var(--background-start-rgb)));
  }
}

body {
  color: rgb(var(--foreground-rgb));
  background: linear-gradient(
      to bottom,
      transparent,
      rgb(var(--background-end-rgb))
    )
    rgb(var(--background-start-rgb));
}

* {
  animation: fade-in 0.5s ease-in-out;
}

@keyframes loading {
  to { transform: translateX(100%); }
}

@keyframes fade-in {
  from { opacity: 0; }
  to { opacity: 1; }
}

@media only screen and (max-width: 1440px) {
  :root {
    font-size: calc(100vw * 16 / 1440);
  }
}

@media only screen and (max-width: 768px) {
  :root {
    font-size: calc(100vw * 16 / 768);
  }
}
