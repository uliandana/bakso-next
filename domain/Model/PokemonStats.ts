export interface PokemonStats {
  name: string,
  value: number,
}
