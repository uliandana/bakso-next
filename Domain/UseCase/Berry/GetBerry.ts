import { Berry } from '@/Domain/Model/Berry';
import { BerryRepository } from '@/Domain/Repository/BerryRepository';

export interface GetBerryUseCase {
  invoke: () => Promise<Berry[]>,
}

export class GetBerry implements GetBerryUseCase {
  private berryRepo: BerryRepository;
  
  constructor(_berryRepo: BerryRepository) {
    this.berryRepo = _berryRepo;
  }

  async invoke() {
    return this.berryRepo.getBerry();
  }
}
