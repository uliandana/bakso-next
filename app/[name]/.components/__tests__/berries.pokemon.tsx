import React from 'react';
import { describe, expect, jest, test } from '@jest/globals';
import Spinner from '@/app/.elements/Spinner';
import { Berry } from '@/domain/Model/Berry';
import ShallowRenderer from 'react-test-renderer/shallow';
import Berries from '../berries.pokemon';

const dummyBerry: Berry = {
  id: '1',
  name: 'banana',
  sprite: 'url',
  firmness: 'soft',
};

describe('./app/[name]/.components/berries.pokemon', () => {
  test('renders', () => {
    const shallow = new ShallowRenderer();
    const onSelect = jest.fn();
    const tree = shallow.render(<Berries isFetchingBerry={false} berries={[dummyBerry]} onSelect={onSelect} />);
    expect(tree).toMatchSnapshot();
  });

  test('render spinner', () => {
    const onSelect = jest.fn();
    const props = {
      isFetchingBerry: true,
      berries: [dummyBerry],
      onSelect,
    };
    const res = Berries(props);
    expect(res.type).toBe(Spinner);
  });

  test('render bg white', () => {
    const onSelect = jest.fn();
    const props = {
      isFetchingBerry: false,
      berries: [{ ...dummyBerry, firmness: '' }],
      onSelect,
    };
    const res = Berries(props);
    expect(res.props.children[0].props.style.backgroundColor).toBe('white');
  });
});
