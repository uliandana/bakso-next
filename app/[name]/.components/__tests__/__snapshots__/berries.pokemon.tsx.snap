// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./app/[name]/.components/berries.pokemon renders 1`] = `
<div
  className="berries"
>
  <label
    htmlFor="berry-1"
    style={
      {
        "backgroundColor": "#775544",
      }
    }
  >
    <input
      className="hidden"
      id="berry-1"
      name="berry"
      onChange={[MockFunction]}
      type="radio"
      value="1"
    />
    <img
      alt="banana"
      className="size-[3rem] m-[1rem]"
      src="url"
      title="banana"
    />
  </label>
</div>
`;
